client/my-sites/sharing/connections/services/mailchimp.js

```diff
@@ -0,0 +1,86 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import PropTypes from 'prop-types';
+import { last, isEqual } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { deleteStoredKeyringConnection } from 'state/sharing/keyring/actions';
+import { SharingService, connectFor } from 'my-sites/sharing/connections/service';
+
+export class Mailchimp extends SharingService {
+	static propTypes = {
+		...SharingService.propTypes,
+		deleteStoredKeyringConnection: PropTypes.func,
+	};
+
+	static defaultProps = {
+		...SharingService.defaultProps,
+		deleteStoredKeyringConnection: () => {},
+	};
+
+	createOrUpdateConnection = () => {};
+
+	/**
+	 * Deletes the passed connections.
+	 *
+	 * @param {Array} connections Optional. Connections to be deleted.
+	 *                            Default: All connections for this service.
+	 */
+	removeConnection = () => {
+		this.setState( { isDisconnecting: true } );
+		this.props.deleteStoredKeyringConnection( last( this.props.keyringConnections ) );
+	};
+
+	componentWillReceiveProps( { availableExternalAccounts } ) {
```

Comment:
`componentWillReceiveProps` is deprecated. It should be converted into a combination of static `getDerivedStateFromProps` that updates state based on the incoming props changes, and `componentDidUpdate`. But that's way out of scope of this PR. All the components based on `SharingService` need to be migrated at the same time.